main.rb

```diff
@@ -0,0 +1,168 @@
+require './book'
+require './person'
+require './student'
+require './rental'
+require './teacher'
+
+@books_arr = []
+@people_arr = []
+@rentals_arr = []
+
+def book_store(book)
+  @books_arr.push(book)
+  @books_arr
+end
+
+def people_store(person)
+  @people_arr.push(person)
+  @people_arr
+end
+
+def rentals_store(rental)
+  @rentals_arr.push(rental)
+  @rentals_arr
+end
+
+def success(message)
+  puts "Successfully created #{message}"
+end
+
+def user_info
+  print 'Enter the name of the person: '
+  # rubocop:disable Lint/UselessAssignment
+  name = gets.chomp
+  print 'Age: '
+  age = gets.chomp
+  # rubocop:enable Lint/UselessAssignment
+end
+
+def create_student
+  user_info
+  print 'Has parent permission? [Y/N]: '
+  parent_permission = gets.chomp
+  case parent_permission.upcase
+  when 'Y'
+    permission = true
+  when 'N'
+    permission = false
+  else
+    puts 'Invalid input'
+    run
+  end
+  student = Student.new(age, name, parent_permission: permission)
+  student.type = 'Student'
+  person_store(student)
+  success('Student')
+end
+
+def create_teacher
+  user_info
+  puts 'Enter the specialization: '
+  specialization = gets.chomp
+  teacher = Teacher.new(specialization, age, name)
+  teacher.type = 'Teacher'
+  person_store(teacher)
+  success('Teacher')
+end
+
```

Comment:
- [ ] Please I am not able to create a `person`(student or teacher), or `book` , please checkout the screenshot below for error messages and kindly fix that, since it is a requirement for this project.
![School_library_UI](https://user-images.githubusercontent.com/65852621/158358028-35f234c6-2265-4583-b878-2c05461884d3.png)